core/src/test/java/org/web3j/protocol/RequestTester.java

```diff
@@ -42,7 +46,12 @@ protected void verifyResult(String expected) throws Exception {
 
         Buffer buffer = new Buffer();
         requestBody.writeTo(buffer);
-        assertThat(buffer.readUtf8(), is(expected));
+        org.web3j.protocol.core.Request actualRequest =
```

Comment:
Unfortunately, I'm not keen on this approach.

We shouldn't be deserialising back into an object - the equality comparison should be using strings.

What I'd suggest is that you replace the `id: <generatedValue>}` via a regex that replaces this with `id: <generatedValue>}` then you check for string equality.

I think it's important that the JSON-RPC requests we're testing are similar to what's being sent down the wire.

Reply:
I'll fix this

Reply:
I realised, that I don't really understand what kind of fix are you talking about. I see 2 solutions, both with their cons and pros:
1. use any long number in test method arguments like this `verifyResult("...., \"id\":42")`, replace this inside `verifyResult` method with some phrase(like `<generatedValue>`), replace actual request id with same phrase and compare strings. This one means you should pass some magic constant, which might be not very obvious for those who will read this.
2. pass some code phrase in test methods instead of magic constant: `verifyResult("...., \"id\":<generatedValue>")`, replace actual request id with same code phrase and compare strings. No magic constant, but makes it a little harder to pass correct value, and it's still not very obvious. Writing new test will probably lead to some research - what should I copy-paste and why is it works.

What are your thoughts about it?

Reply:
Hey @IgorPerikov, sorry for the delay in coming back to you - just got back from Devcon.

Option 1 is in line with my thinking. 

Thanks.